server/netty-server/cats/src/main/scala/sttp/tapir/server/netty/cats/internal/WebSocketPipeProcessor.scala

```diff
@@ -98,10 +100,19 @@ class WebSocketPipeProcessor[F[_]: Async, REQ, RESP](
 
   }
 
-  private def optionallyConcatenateFrames(s: Stream[F, WebSocketFrame], doConcatenate: Boolean): Stream[F, WebSocketFrame] =
+  private def optionallyConcatenateFrames(doConcatenate: Boolean)(s: Stream[F, WebSocketFrame]): Stream[F, WebSocketFrame] =
     if (doConcatenate) {
       s.mapAccumulate(None: Accumulator)(accumulateFrameState).collect { case (_, Some(f)) => f }
     } else s
+
+  private def takeUntilCloseFrame(passAlongCloseFrame: Boolean)(s: Stream[F, WebSocketFrame]): Stream[F, WebSocketFrame] =
+    s.takeWhile(
+      {
+        case _: WebSocketFrame.Close => false
+        case _                       => true
+      },
+      takeFailure = passAlongCloseFrame
```

Comment:
hm this (`takeFailure`) looks weird - we should find a better name ;) this is only for the *first* frame that fails the test, right?

Reply:
Yes. It's how they called it in `fs2`. My first version was named `takeLast`, but I wasn't happy with it (takeLast = false looks like it skips the failing element, so takes even less?). ZIO has a `takeUntil` method to deal with this.
Akka streams has `inclusive: Boolean` and this looks better than `takeFailure`, wdyt?

Reply:
`includeFirstFailing`?